robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowApplication.java

```diff
@@ -833,4 +836,21 @@ public void setSystemService(String key, Object service) {
     shadowContext.setSystemService(key, service);
   }
 
+  @Implementation
+  public void registerActivityLifecycleCallbacks(Application.ActivityLifecycleCallbacks callback) {
```

Comment:
Yeah, that's a good point. Also, I've noticed that I can use a mock Application in the tests that are using these implementation methods. I'm happy to drop this PR if it's not generally useful.